Hardcode a.out related constants

The OMAGIC, NMAGIC and ZMAGIC constants, related to the legacy a.out
binary format, are no longer available in the headers of all
architectures, due to this binary format being deprecated. For that
reason, util-linux fails to build.

We therefore take the approach used in newer versions of
util-linux-ng: hardcoding those values. See
http://git.kernel.org/?p=utils/util-linux-ng/util-linux-ng.git;a=commitdiff;h=e9eb4dad0ece03914f67f6d1ded74cfbbe694ba4

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 text-utils/more.c |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

Index: util-linux-2.13-pre7/text-utils/more.c
===================================================================
--- util-linux-2.13-pre7.orig/text-utils/more.c
+++ util-linux-2.13-pre7/text-utils/more.c
@@ -53,7 +53,6 @@
 #include <sys/stat.h>
 #include <sys/file.h>
 #include <sys/wait.h>
-#include <a.out.h>
 #include <locale.h>
 #include "xstrncpy.h"
 #include "nls.h"
@@ -504,9 +503,9 @@
 
 	if (fread(twobytes, 2, 1, f) == 1) {
 		switch(twobytes[0] + (twobytes[1]<<8)) {
-		case OMAGIC:	/* 0407 */
-		case NMAGIC:	/* 0410 */
-		case ZMAGIC:	/* 0413 */
+		case 0407:	/* a.out obj */
+		case 0410:	/* a.out exec */
+		case 0413:	/* a.out demand exec */
 		case 0405:
 		case 0411:
 		case 0177545:
